src/btree/bt_cursor.c

```diff
@@ -72,11 +72,25 @@ __cursor_page_pinned(WT_CURSOR_BTREE *cbt)
     }
 
     /*
-     * Check if the key references the page. When returning from search, the page is active and the
-     * key is internal. After the application sets a key, the key is external, and the page is
-     * useless.
+     * Check if the key references an item on a page. When returning from search, the page is pinned
+     * and the key is internal. After the application sets a key, the key becomes external. For the
+     * search and search-near operations, we assume locality and check any pinned page first on each
+     * new search operation. For operations other than search and search-near, check if we have an
+     * internal key. If the page is pinned and we're pointing into the page, we don't need to search
+     * at all, we can proceed with the operation. However, if the key has been set, that is, it's an
+     * external key, we're going to have to do a full search.
      */
-    if (!F_ISSET(cursor, WT_CURSTD_KEY_INT))
+    if (!search_operation && !F_ISSET(cursor, WT_CURSTD_KEY_INT))
+        return (false);
+
+    /*
+     * XXX No fast-path searches at read-committed isolation. Underlying transactional functions
+     * called by the fast and slow path search code handle transaction IDs differently, resulting in
+     * different search results at read-committed isolation. This makes no difference for the update
+     * functions, but in the case of a search, we will see different results based on the cursor's
+     * initial location. See WT-5134 for the details.
```

Comment:
It's generally OK, since our bug tracking is public the information is available to everyone.
As in this example, the comment should give as much context as is practical inline, so it's only really for cases where there is a lot of subtlety or a non-obvious tradeoff.

Reply:
Ok, thanks for the details. I also agree to mention the Jira ticket with this case. Going through the Jira ticket will provide more details.